src/pipeline.ts

```diff
@@ -489,14 +541,507 @@ export async function scanRssFeeds() {
+		const result = await getClient().isTorrentComplete(meta.infoHash);
+		let isComplete = result.isOk() ? result.unwrap() : false;
+		if (isComplete) {
+			toDelete.add(torrentFilePath);
+		}
+		if (injectionResult === InjectionResult.ALREADY_EXISTS) {
+			if (linkedNewFiles) {
+				logger.info({
+					label: Label.INJECT,
+					message: `${progress} Rechecking ${filePathLog} as new files were linked - ${chalk.green(injectionResult)}`,
+				});
+				toRecheck.add(meta.infoHash);
+			} else if (fullMatches.length && !isComplete) {
+				logger.info({
+					label: Label.INJECT,
+					message: `${progress} Rechecking ${filePathLog} as it's not complete but has all files - ${chalk.green(injectionResult)}`,
+				});
+				isComplete = true;
+				toRecheck.add(meta.infoHash);
+				toDelete.add(torrentFilePath); // Prevent infinite recheck in rare case of corrupted cross seed
+			} else {
+				logger.warn({
+					label: Label.INJECT,
+					message: `${progress} Unable to inject ${filePathLog} - ${chalk.yellow(injectionResult)}${isComplete ? "" : " (incomplete)"}`,
+				});
+			}
+			totalAlreadyExists++;
+			totalCandidateIncomplete += isComplete ? 0 : 1;
+			continue;
+		}
+		logger.info({
+			label: Label.INJECT,
+			message: `${progress} Injected ${filePathLog} - ${chalk.green(injectionResult)}`,
+		});
+		sendResultsNotification(matchedSearchee!, [
+			[
+				{ decision: matchedDecision!, metafile: meta },
+				tracker,
+				injectionResult,
+			],
+		]);
+		totalInjected++;
+		if (matchedDecision! === Decision.MATCH_PARTIAL) {
+			totalPartialMatches++;
+		} else {
+			totalFullMatches++;
+		}
+	}
+
+	for (const infoHash of toRecheck) {
+		await getClient().recheckTorrent(infoHash);
+	}
+
+	const incompleteMsg = `${chalk.bold.yellow(totalAlreadyExists)} existed in client${
+		totalCandidateIncomplete
+			? chalk.dim(` (${totalCandidateIncomplete} were incomplete)`)
+			: ""
+	}`;
+	const resultMsg = formatAsList(
+		[
+			`Injected ${chalk.bold.green(totalInjected)}/${chalk.bold.white(dirContents.length)} torrents`,
+			totalFullMatches &&
+				`${chalk.bold.green(totalFullMatches)} were full matches`,
+			totalPartialMatches &&
+				`${chalk.bold.yellow(totalPartialMatches)} were partial matches`,
+			totalSearcheeIncomplete &&
+				`${chalk.bold.yellow(totalSearcheeIncomplete)} had incomplete sources`,
+			totalAlreadyExists && incompleteMsg,
+			totalBlocked &&
+				`${chalk.bold.yellow(totalBlocked)} were possibly blocklisted`,
+			totalFailed && `${chalk.bold.red(totalFailed)} failed to inject`,
+			totalUnmatched &&
+				`${chalk.bold.red(totalUnmatched)} had no matches`,
+		].filter(isTruthy),
+		{ sort: false, type: "unit" },
+	);
+	logger.info({ label: Label.INJECT, message: chalk.cyan(resultMsg) });
+
+	if (totalUnmatched > 0) {
+		logger.info({
+			label: Label.INJECT,
+			message: `Use "${chalk.bold.white("cross-seed diff")}" to get the reasons two torrents are not considered matches`,
+		});
+	}
+
+	if (foundBadFormat && !flatLinking) {
```

Comment:
there's a `logOnce` helper https://github.com/cross-seed/cross-seed/blob/4cb1f50e75644dfaf9e99b48eab40e16c06a2996/src/logger.ts#L87 that might allow us to "fire and forget" and not worry about storing this information in `foundBadFormat`.

Reply:
I want these to be logged at the end with all the summary, it would cause this to be lost in the middle of the logs.